### PR TITLE
Replacement of D2K mod static shellmap with a scripted shellmap

### PR DESCRIPTION
**Pull request by Mix-martes86**
Complete replacement of the fixed-screen shellmap currently in use for a customized shellmap, taking original RA mod shellmap as basis. WIP, terrain is not 100% finished, needs corrections, some buildings on sand instead of rock, and repetitive terrain patterns. Some scripting fixes done when rules for structures were modified. Commit 0cc324e. Reference (don't mind the music, used D2K's score track in script): https://www.youtube.com/watch?v=8Ioku0YT2F0

### CODE REVIEW

**Comment by penev92** [comment]
Sorry, but while I'd welcome a more interesting shellmap, I don't think that a direct copy of RA's shellmap is a good idea:
 - The base layouts, while OK for RA, don't make much sense for D2k.
 - The map should highlight some D2k-specific features (carryalls, sandworms, starport delivery, Deviator tanks). Maybe have more emphasis on harvesting/protecting harvesters and the limited buildable space.

**Comment by Mix-martes86** [comment]
The idea for this shellmap is a year old (I just learned about Git basics and some more things, had no time until my recent unemployment), so obviously it has some holes.

My intention was to replace the, back then, completely empty and lifeless default shellmap on the release builds, at least temporarily, until an improved version came out, seeing how certain features can understandably be quite low in the priority queue. Or at least, that it could serve as sort of a template or tool for someone to make something nicer. And because I was long overdue since I spoke to people about it, I pushed it as is.

I thought about starport deliveries, but since I didn't know how to script them, I just left it as todo stuff. Carryalls are however in the map. Last release seems to automatically add them to harvesters without me doing anything. And sandworms, from what I've seen, can be ruled in as well without much hassle.

In any case, I'll see what can I do. I used a custom in-game editor (not done by me) back then, which considerably eased the tediousness of placing stuff or drawing the terrain, but I can't find the source.

**Comment by penev92** [comment]
The in-game map editor was actually merged last week, so you can find it in Main menu -> Extras -> Map editor.
Should make things much easier.
I am not convinced about the map/bases layout though.

File: mods/d2k/maps/shellmap/map.yaml [hunk 1]
```diff
@@ -33,102 +33,1571 @@ Players:
...
 		-SpawnMPUnits:
 		-MPStartLocations:
 		ResourceType@Spice:
 			ValuePerUnit: 0
-		WormManager:
-			Minimum: 1
-			Maximum: 1
+		LoadWidgetAtGameStart:
+			Widget: MAINMENU
+		-StartGameNotification:
```
Comment (Mailaender) [inline mods/d2k/maps/shellmap/map.yaml]: This triggers a lint rule error. https://travis-ci.org/OpenRA/OpenRA/builds/63595331#L1176

File: mods/d2k/maps/shellmap/map.yaml [hunk 2]
```diff
@@ -33,102 +33,1571 @@ Players:
...
 		-CrateSpawner:
 		-SpawnMPUnits:
 		-MPStartLocations:
 		ResourceType@Spice:
 			ValuePerUnit: 0
-		WormManager:
-			Minimum: 1
-			Maximum: 1
+		LoadWidgetAtGameStart:
+			Widget: MAINMENU
```
Comment (Mailaender) [inline mods/d2k/maps/shellmap/map.yaml]: You don't need this anymore. We removed it during the preparation of the in-game map editor. https://github.com/OpenRA/OpenRA/pull/7217

File: mods/d2k/maps/shellmap/map.yaml (hunk 1 above)
Comment (Mix-martes86) [inline mods/d2k/maps/shellmap/map.yaml]: Yeah, noticed that when I made the request. I'll fix it in next commit.

File: mods/d2k/maps/shellmap/map.yaml (hunk 2 above)
Comment (Mix-martes86) [inline mods/d2k/maps/shellmap/map.yaml]: Noted. On for the next commit.

Comment (Mailaender) [inline mods/d2k/maps/shellmap/map.yaml]: No, not the next commit. Use git amend and rebase to fix your previous commits.

**Comment by Mix-martes86** [comment]
I'm looking at this rebase thing, but I have absolutely no idea what I'm doing. I might break the commit history.

**Comment by Mix-martes86** [comment]
I give up, I don't want to break the origin repo by accident with this little experiment. This damned Git is the least user-friendly thing ever, I keep having to use a shell console to consolidate conflicts with already commited stuff, and I don't have the patience for that. I'll keep looking at improving the shellmap and upload it for someone to copy the code and do a pull request themselves if it raises any interest.

**Comment by deniz1a** [comment]
You can try https://www.sourcetreeapp.com/ . It's user friendly. Also if you have problems with rebasing just clone a fresh copy of bleed and do your changes there...

**Comment by Mix-martes86** [comment]
That app is definitely friendlier looking, thanks. Will give it another try and see what happens.

**Comment by x-a-n-a-x** [comment]
SmartGit is more stable and faster (in my opinion) but I agree with Deniz1a, don't give up.
 http://www.syntevo.com/smartgit/

**Pull request closed without merging**